Elements: Buttons

clrux opened the pull request:
# Elements: Buttons
This pull request establishes Sass partials for our buttons and contains a demo page that displays them all using our created styles.

**Buttons with icons are not yet included as we don't have a finished set of icons to incorporate.**

clrux commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+
+description:  Examples of the different types and variants of edX buttons.
+
+info: Buttons are great. Mkay.
```
Need to update this.

talbs replied:
lol, yes, but please leav the "Mkay" in when you do (j/k).

clrux commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+    <button type="button" class="btn btn-primary btn-base btn-combo">Combo Base</button>
+</section>
+
+<!--<section class="example-set">
```
Until we get the icons popped in. Or would you prefer we just remove this for now @talbs?

talbs replied:
Nahh, this is a good thing to set up and the class names make a lot of sense.

clrux commented on _src/static/sass/utilities/_variables.scss:
```diff
@@ -322,6 +322,7 @@ $timing: (
     x-slow:     1.5s,
     fast:       0.5s,
     x-fast:     0.25s,
+    xx-fast:    0.1s
```
Oh yes I did.

talbs replied:
Fast and the Furious 8: CSS Transitions (cc: @frrrances)

talbs commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+info: Buttons are great. Mkay.
+---
+
+<section class="example-set">
```
This is a nice convention. Can we sync up other PL posts with these ``<section class="example-set">`` containers if they are not already (in a separate PR from this work)?

talbs commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+---
+
+<section class="example-set">
+    <h3>Default</h3>
```
NIT: can we use our headings pattern classes here to render a specific heading style/level?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+// ----------------------------
+// #LAYOUT
+// ----------------------------
+.example-set {
```
Hmm, this shouldn't be in the component/element rendering. A better place is in the ``/src/pl/static/...`` Sass, since this is styling that is just to render these buttons in the preview/doc site. Please move it to there (if its not already there).

Oh, and if you're following bad form/habits from some previous work, mind removing any of the same from other component Sass?

talbs replied:
@clrux, can you move this given my prior comment?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    margin-bottom: spacing-vertical(base);
+
+    .button-overlay-demo {
+        padding: 2rem;
```
See above note, also from a syntax PoV, should this be ``rem(2)``?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+// ----------------------------
+.btn {
+    @include transition(
+        color timing(xx-fast) ease-in-out 0s,
```
Nice syntax with the line breaks between properties here.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    );
+    display: inline-block;
+    border-style: solid;
+    border-radius: 3px;
```
Philosophical question: is it okay to have these values be in ``px`` rather than ``rems`` or fluid units? They seem defined the easiest this way, but are there any drawbacks when it comes to scaling UI up or down by having them locked in?

clrux replied:
Good question. I was actually wondering about this. Pixels allows us to be precise. Not to say that using `rems` wouldn't allow precision, because it's based on a globally set variable, but the pixel conversion would be something like 2.57945 or 3.172938 (as examples) and may not look exactly how we want. If you're okay with that inexactness, I'm okay using `rems` rather than `px` for this.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+// ----------------------------
+.btn-base {
+    font-size: font-size(mid-large);
+    padding: .8rem 1.2rem;
```
Should this be using the Bourbon ``rem()`` syntax rather than raw rems?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+// ----------------------------
+.btn-large {
+    font-size: font-size(large);
+    padding: 1rem 1.4rem;
```
Should this be using the Bourbon ``rem()`` syntax rather than raw rems?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+// ----------------------------
+.btn-small {
+    font-size: font-size(small);
+    padding: .6rem 1rem;
```
Should this be using the Bourbon ``rem()`` syntax rather than raw ``rem``s?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: palette(primary, mid-dark);
+    color: palette(primary, mid-dark);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Could you elaborate here?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(grayscale, white);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
And here.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-color: palette(primary, dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
Could you elaborate on the first note?

As for the second, if we're using proper, semantic markup, adding `disabled` to the button (because we should be using a button for buttony things) should be enough. Can you give me an example of where this might be beneficial?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: palette(grayscale-cool, base);
+    color: palette(grayscale-cool, base);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(grayscale, white);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-color: palette(primary, dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
See my comment(s) above.

clrux replied:
If we're using proper, semantic markup, adding disabled to the button (because we should be using a button for buttony things) should be enough. Can you give me an example of where this might be beneficial?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: palette(primary, mid-dark);
+    color: palette(grayscale, white);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-color: palette(primary, base);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-color: palette(primary, mid-dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
See my comment(s) above.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: $transparent;
+    color: palette(primary, base);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-color: palette(grayscale-cool, trans);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(primary, mid-dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
See my comment(s) above.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: $transparent;
+    color: palette(grayscale-cool, base);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(grayscale, white);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-color: palette(primary, dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
See my comment(s) above.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: $transparent;
+    color: palette(primary, mid-dark);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        text-decoration: underline;
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(primary, dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
See my comment(s) above.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    border-color: palette(primary, dark);
+    color: palette(grayscale, white);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        box-shadow: inset 0 -3px 0 palette(primary, mid-dark);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        box-shadow: inset 0 3px 0 palette(primary, x-dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs replied:
Is it worth throwing any attribute selectors in here for cases like ``<button type="button" disabled>...</button>``?

clrux replied:
See my comment(s) above.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+.btn-combo {
+    border-radius: 0;
+
+    &:first-of-type {
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        border-bottom-left-radius: 3px;
+    }
+
+    &:last-of-type {
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+    color: palette(grayscale, white);
+    font-weight: font-weight(semi-bold);
+
+    &:hover,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(primary, mid-dark);
+    }
+
+    &:active,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,371 @@
+        color: palette(primary, x-dark);
+    }
+
+    &:disabled,
```
NIT: add stateful Sass comments - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

clrux replied:
Can you elaborate?

talbs commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+tags:
+- atomic
+- pattern
+- headings
```
Please fix to "buttons"

talbs commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+- headings
+
+slug:         buttons
+scssurl:      "https://github.com/edx/ux-pattern-library/blob/master/_src/static/sass/components/_buttons.scss"
```
Please double-check that this is where the .scss file will be.

clrux replied:
Yes?

talbs commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,97 @@
+slug:         buttons
+scssurl:      "https://github.com/edx/ux-pattern-library/blob/master/_src/static/sass/components/_buttons.scss"
+
+description:  Examples of the different types and variants of edX buttons.
```
Need to update this as well.

talbs commented:
@clrux, this is really well-organized work. Thanks!

I've left some comments inline, but I also have a few larger ones for all of buttons:

1. **Theming** - I'd like us to plumb through theming into the values of a lot of these button properties . Remember that we're trying to tackle theming hooks with each new pattern/component we're adding. Can you A) revise and abstract the properties you've added based on the [spirit of this Theming inventory doc](https://docs.google.com/a/edx.org/spreadsheets/d/1kHttF70rf2p7-9-Ev6kubfSFwncrDi0KzwFO9jArOx0/edit?usp=sharing), B) adjust anything needed at a configuration level (mostly in ``_variables.scss``) to support that work, and C) test this against an updated version of your faux MIT theming branch?
2. **Arrangement + Alignment** - While grids are a bit away, should we give folks utility classes to have buttons display inline/inline-block next to each other vs. taking up a whole visual line with more block displays
3. **Links and Other Actions** - The examples you added are all ``<button>``-based. These classes should cosmetically be able to be applied to other things we want to visually look like an action (the most common case will be an ``<a>``). Can you test that this can be done and also provide an example of that in your new post markdown? Also, is this the time and place to add basic text link (and maybe image/whole block-level element) link stylng since you have some visual cues to go from?

- - -

Let me know on all of the above. I'm still reviewing the actual rendering/visual design and will try to get some for the group to review the buttons now that you've put em in the browser wild.

clrux commented on _src/pl/static/sass/_views.scss:
```diff
@@ -30,3 +30,18 @@
+// ------------------------------
+// #TAB PANELS / EXAMPLES
+// ------------------------------
+.example-set {
```
Added a globally available wrapper for examples.

talbs replied:
I think this stuff is better defined in the ``_components.scss`` partial in the same directory. Its less of a view and more components on PL doc site specific views.

clrux commented on _src/pl/static/sass/_views.scss:
```diff
@@ -30,3 +30,18 @@
+    border-bottom: spacing-v(xx-small) solid palette(grayscale, xx-light);
+    margin-bottom: spacing-v(mid-large);
+
+    .button-overlay-demo {
```
Temporary for the demo. Should not be included outside this site.

clrux commented:
@talbs I've gone through and addressed your comments. As for your three additional, global items, I've:
* piped variables through (and created new variables as needed) to make this more themeable
* created a `blocked` class which presents `.btn`'s as blocked and full-width of their parent container.
* added some link examples that use these classes; all looks good here
* attempted to get these to work with the MIT theme, but for some reason it's not seeing the added styles. Let's touch base on this on Monday when you return.

If you have time between now and then, feel free to take another pass.

talbs commented on _src/static/sass/_build.scss:
```diff
@@ -23,6 +23,7 @@
 // ------------------------------
 @import 'components/headings';
 @import 'components/colors';
+@import 'components/buttons';
```
Yes! One more closer to world domination!

talbs commented on _src/static/sass/utilities/_variables.scss:
```diff
@@ -362,6 +363,12 @@ $headings-primary-color: palette(grayscale, x-dark);
 $headings-secondary-font-family: $font-family-sans-serif;
 $headings-secondary-color: palette(grayscale, base);
 
+// buttons
```
This is a good start to theming settings, but I don't see any of the following:

* text color (inactive, focus/hover, active, disabled)
* background color (inactive, focus/hover, active, disabled)
* settings for specific levels of buttons

clrux replied:
@talbs Aren't these done in the `_buttons.scss` file, which institutions would then overwrite. I figured the palettes (and some other configuration things) are in `_variables.scss` but that styles specific to buttons would be in that respective partial. Let me know as I don't think it's good to have button configurations in two different places.

`_variables.scss` has global button configuration, such as border style, shading, radius
`_buttons.scss` has specific button configuration, such as background/foreground color, border color, etc.

This okay with you?

talbs replied:
Yeah, after thinking about it more, that approach makes a lot of sense. Thanks!

talbs commented:
Also, @clrux, you need to rebase to get rid of the extra commit that's already been merged (https://github.com/edx/ux-pattern-library/commit/fa1e7d67d2565306fc13b439c2680c9da3b10b8e).

talbs commented on _posts/elements/2015-04-09-buttons.md:
```diff
@@ -0,0 +1,105 @@
+
+description:  A collection of buttons available for using within the edX platform.
+
+info: Buttons should be used for performing actions within the edX environment. While we supply a button that looks like a link, it should only really be used for very tertiary _actions_. edX offers three button sizes, each with normal, hover, active/pressed, and disabled states. We also offer a range of other buttons to suit your needs.
```
NIT: I don't think this paragraph is being parsed for markdown so the ``_actions_`` formatting isn't being applied. Mind removing it?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,389 @@
+// ----------------------------
+.btn-base {
+    font-size: font-size(mid-large);
+    padding: rem(($baseline/1.5)) rem(($baseline*1.25));
```
Should we be using ``spacing-vertical()`` and `spacing-horizontal()`` values instead of baseline here? I think there are merits to us staying with one way to reference our set scales of vertical/horizontal spacing.

clrux replied:
@talbs I've updated the `$baseline` units to utilize our `spacing-vertical` and `spacing-horizontal` formulas. The sizing isn't exactly as you designed though. We can adjust the scales which may impact other things, or we can use what I previously had. Let me know.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,389 @@
+// ----------------------------
+.btn-large {
+    font-size: font-size(large);
+    padding: rem($baseline) rem(($baseline*1.5));
```
Should we be using ``spacing-vertical()`` and `spacing-horizontal()`` values instead of baseline here? I think there are merits to us staying with one way to reference our set scales of vertical/horizontal spacing.

talbs replied:
See [my spacing comment above](https://github.com/edx/ux-pattern-library/pull/19/files#r28337162).

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,389 @@
+// ----------------------------
+.btn-small {
+    font-size: font-size(small);
+    padding: rem(($baseline/2)) rem(($baseline/1.5));
```
See [my spacing comment above](https://github.com/edx/ux-pattern-library/pull/19/files#r28337162).

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,389 @@
+    }
+
+    // STATE: is disabled
+    &:disabled,
```
I'm still getting a hover state visually when previewing this. Can you make sure this is not the case?

talbs replied:
Also, is this combinations of BG/FG colors a11y-friendly?

clrux replied:
Probably not. Let me review and make tweaks as needed. Might impact our scale/palettes. That okay?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,389 @@
+    // STATE: is disabled
+    &:disabled,
+    &.is-disabled {
+        display: none; // :)
```
Haha! Do you really want to keep this in or can it be removed?

clrux replied:
Doesn't matter to me. I was basically saying that links don't really have disabled states, so if we're using an inactive link (or a link-button), we should just not show it in the UI. Like, how do you make a link look inactive without making it look like text? You could fade it back, but then you risk running into a11y issues for contrast.

talbs commented on _src/static/sass/utilities/_variables.scss:
```diff
@@ -362,6 +363,12 @@ $headings-primary-color: palette(grayscale, x-dark);
+$buttons-border-style: solid;
+$buttons-border-size: 1px;
+$buttons-border-radius: 3px;
+$buttons-shading: 3px;
```
This value is really only used in our defined buttons with ``.btn-elevated``. Should this variable be renamed to show that scope (i.e. ``.btn-elevated-shading``)?

talbs replied:
Also, this is less shading than it is a shadow. Let's make that connection more direct with a ``.btn-shadow`` (or ``.btn-elevated-shadow`` if you agree with my previous comment).

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,398 @@
+        color: palette(primary, mid-dark);
+    }
+
+    // STATE: is disabled
```
Is this combinations of BG/FG colors a11y-friendly?

talbs replied:
Can this be inherited from ``.btn-default``?

clrux replied:
They look a little different to me so not sure they can use the same styles.

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,398 @@
+    }
+
+    // STATE: is disabled
+    &:disabled,
```
Is this combinations of BG/FG colors a11y-friendly?

talbs commented on _src/static/sass/components/_buttons.scss:
```diff
@@ -0,0 +1,398 @@
+        border-color: palette(primary, dark);
+    }
+
+    // STATE: is disabled
```
Is this combinations of BG/FG colors a11y-friendly?

talbs replied:
Can this be inherited from ``.btn-default``?

clrux replied:
Same as above.

talbs replied:
Gotcha.

talbs commented:
@clrux, thanks for the good work - this is looking great. I've placed the rest of my comments inline on this PR and also made https://github.com/edx/ux-pattern-library/pull/45 for some quick clean up tasks. Mind reviewing that and signing off to merge into here?

Once all the dust has settled and you've rebased from master, :+1:

clrux commented:
@talbs Values are a11y friendly!

talbs commented:
Sweet, thanks! Let me know on my PR and we can move this along.

talbs commented:
Please squash commits. Nice work and thanks for the collaboration!

:+1:
